packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -439,14 +566,10 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             //repay the flash swap
             IWPowerPerp(wPowerPerp).transfer(wPowerPerpPool, _amountToPay);
 
-            if (address(this).balance > 0) {
-                payable(_caller).sendValue(address(this).balance);
-            }
-
             // this is a newly open vault, transfer to the user
             if (data.vaultId == 0) IShortPowerPerp(shortPowerPerp).safeTransferFrom(address(this), _caller, vaultId);
-        } else if (FLASH_SOURCE(_callSource) == FLASH_SOURCE.FLASH_W_BURN) {
-            FlashWBurnData memory data = abi.decode(_callData, (FlashWBurnData));
+        } else if (CALLBACK_SOURCE(_callSource) == CALLBACK_SOURCE.FLASH_W_BURN) {
```

Comment:
we have to send back extra ETH to the user if collateralToWithdraw is > amount paid for buying back